api/src/identity-access-management/domain/models/ClaimsMapper.js

```diff
@@ -0,0 +1,50 @@
+import flatten from 'lodash/flatten.js';
+import pick from 'lodash/pick.js';
+
+import { DEFAULT_CLAIMS_MAPPING } from '../constants/oidc-identity-providers.js';
+
+const DEFAULT_OPTIONS = { mapping: DEFAULT_CLAIMS_MAPPING, additionalClaims: [] };
+
+export class ClaimsMapper {
+  constructor(options) {
+    const { mapping = {}, additionalClaims = [] } = { ...DEFAULT_OPTIONS, ...(options || {}) };
+
+    this.additionalClaims = additionalClaims;
+
+    this.mapping = {
+      ...mapping,
+      ...additionalClaims.reduce((result, claim) => ({ ...result, [claim]: [claim] }), {}),
+    };
+  }
+
+  map(userInfo = {}) {
```

Comment:
Une remarque pour toutes les fonctions de cette classe : pourquoi pour chacune de ces fonctions fournir une valeur par défaut pour `userInfo` (`userInfo = {}`) ? Cela laisse penser qu'il est légitime d'utiliser ces fonctions sans leur passer de paramètre. Or je n'ai pas l'impression que ce cas arrive dans les cas d'utilisation que nous avons avec l’OidcAuthenticationService. Du coup si cette suggestion est acceptée, on pourra aussi supprimer les tests qui testent les cas où les `userInfo` passés sont `null`.

```suggestion
  map(userInfo) {
```

Reply:
La classe `ClaimManager` est indépendante de `OidcAuthenticationService` ou tout autre classe, et ne sait pas comment elle est utilisée. Ce qui est bien car sa responsabilité est isolée.

On s'assure donc que les fonctions qu'elle expose soient au minimum dans ce cas null-safe. Ainsi peu importe dans quel contexte où ces méthodes sont utilisées il y a moins de risque d'erreur non désirées.

Note: si on était en Typescript, en effet je n'aurais pas géré ce cas car on aurait typé les `userInfo` en `Record<string, string>` non `null` et non `undefined`.